target/cheri-common/op_helper_cheri_common.c

```diff
@@ -880,6 +880,15 @@ target_ulong CHERI_HELPER_IMPL(ctestsubset(CPUArchState *env, uint32_t cb,
     return (target_ulong)is_subset;
 }
 
+target_ulong CHERI_HELPER_IMPL(csetequalexact(CPUArchState *env, uint32_t cb,
+                                              uint32_t ct))
+{
+    const cap_register_t *cbp = get_capreg_0_is_ddc(env, cb);
```

Comment:
I am unsure what is more useful: comparing to CNULL or to DDC. Considering we don't use 0 == ddc very much for RISC-V I'd be tempted to use CNULL (same for the sail PR). This would also match MIPS.

But I don't really mind either way. @jrtc27 ?

Reply:
Using DDC for 0 has always been weird, and IMO should be reserved for cases where CNULL is meaningless but DDC provides useful behaviour (ie CToPtr, CFromPtr and CBuildCap), or where CNULL is unlikely to be very useful (ie CTestSubset). Both could conceivably be useful here, but CNULL is less surprising and since it's a meaningful operation to perform I think we should use CNULL.

Reply:
I just stole this from PeterRugg's implementation in Flute.  It does have the sort of "cute" result that

  * `CSetEqualExact rd, 0, 0` compares DDC and NULL
  * `CSetEqualExact rd, cN, 0` compares cN against NULL
  * `CSetEqualExact rd, 0, cN` compares cN against DDC

The remaining diagonal entries are fairly useless, tho', so maybe the "0, 0" one should be too.  (And the redundancy in the encodings also makes me unhappy, but I guess "half of the capreg*capreg space" isn't very many actual opcodes.)